src/SonarScanner.MSBuild.PreProcessor/Roslyn/RoslynAnalyzerProvider.cs

```diff
@@ -28,109 +28,101 @@
+        Debug.Assert(ruleSet is not null, "Expecting the RuleSet to be created.");
+        Debug.Assert(ruleSet.Rules is not null, "Expecting the RuleSet.Rules to be initialized.");
 
+        var rulesetFilePath = Path.Combine(teamBuildSettings.SonarConfigDirectory, string.Format(deactivateAll ? RulesetFileNameNone : RulesetFileNameNormal, language));
+        logger.LogDebug(Resources.RAP_UnpackingRuleset, rulesetFilePath);
         ruleSet.Save(rulesetFilePath);
 
         return rulesetFilePath;
     }
 
-    private IEnumerable<string> WriteAdditionalFiles(string language, IEnumerable<SonarRule> activeRules)
+    private IEnumerable<string> WriteAdditionalFiles(IEnumerable<SonarRule> activeRules)
     {
-        Debug.Assert(activeRules != null, "Supplied active rules should not be null");
+        Debug.Assert(activeRules is not null, "Supplied active rules should not be null");
 
-        if (TryWriteSonarLintXmlFile(language, activeRules, out var filePath))
+        if (TryWriteSonarLintXmlFile(activeRules, out var filePath))
         {
             Debug.Assert(File.Exists(filePath), "Expecting the additional file to exist: {0}", filePath);
-            return new[] { filePath };
+            return [filePath];
         }
-
-        return Enumerable.Empty<string>();
+        return [];
     }
 
-    private bool TryWriteSonarLintXmlFile(string language, IEnumerable<SonarRule> activeRules, out string sonarLintXmlPath)
+    private bool TryWriteSonarLintXmlFile(IEnumerable<SonarRule> activeRules, out string sonarLintXmlPath)
     {
-        var langDir = Path.Combine(this.teamBuildSettings.SonarConfigDirectory, language);
+        var langDir = Path.Combine(teamBuildSettings.SonarConfigDirectory, language);
         Directory.CreateDirectory(langDir);
 
         sonarLintXmlPath = Path.Combine(langDir, "SonarLint.xml");
         if (File.Exists(sonarLintXmlPath))
         {
-            this.logger.LogDebug(Resources.RAP_AdditionalFileAlreadyExists, language, sonarLintXmlPath);
+            logger.LogDebug(Resources.RAP_AdditionalFileAlreadyExists, language, sonarLintXmlPath);
             return false;
         }
 
-        var content = RoslynSonarLint.GenerateXml(activeRules, this.sonarProperties, language);
-        this.logger.LogDebug(Resources.RAP_WritingAdditionalFile, sonarLintXmlPath);
+        var content = RoslynSonarLint.GenerateXml(activeRules, sonarProperties, language);
+        logger.LogDebug(Resources.RAP_WritingAdditionalFile, sonarLintXmlPath);
         File.WriteAllText(sonarLintXmlPath, content);
         return true;
     }
 
-    private IEnumerable<AnalyzerPlugin> FetchAnalyzerPlugins(string language, IEnumerable<SonarRule> activeRules)
+    private IEnumerable<AnalyzerPlugin> FetchAnalyzerPlugins(IEnumerable<SonarRule> activeRules)
     {
         var partialRepoKeys = ActiveRulesPartialRepoKeys(activeRules);
-        IList<Plugin> plugins = new List<Plugin>();
+        IList<Plugin> plugins = [];
```

Comment:
I don't see a reason for `IList`, it just hides the real thing
```suggestion
        List<Plugin> plugins = new();
```